forc-plugins/forc-doc/src/doc.rs

```diff
@@ -265,14 +248,53 @@ impl ModuleInfo {
     }
     /// The depth of a module as `usize`.
     pub(crate) fn depth(&self) -> usize {
-        self.0.len()
+        self.module_prefixes.len()
     }
     /// Create a new [ModuleInfo] from a vec.
-    pub(crate) fn from_vec(vec: Vec<String>) -> Self {
-        Self(vec)
+    pub(crate) fn from_ty_module(module_prefixes: Vec<String>, attributes: Option<String>) -> Self {
+        Self {
+            module_prefixes,
+            attributes,
+        }
+    }
+    pub(crate) fn preview_opt(&self) -> Option<String> {
+        create_preview(self.attributes.clone())
     }
 }
 
+/// Create a docstring preview from raw html attributes.
+///
+/// Returns `None` if there are no attributes.
+fn create_preview(raw_attributes: Option<String>) -> Option<String> {
+    const MAX_PREVIEW_CHARS: usize = 100;
+    const CLOSING_PARAGRAPH_TAG: &str = "</p>";
+
+    raw_attributes.as_ref().map(|description| {
+        let preview = split_at_markdown_header(description);
+        if preview.chars().count() > MAX_PREVIEW_CHARS && preview.contains(CLOSING_PARAGRAPH_TAG) {
+            match preview.find(CLOSING_PARAGRAPH_TAG) {
+                Some(closing_tag_index) => {
+                    // We add 1 here to get the index of the char after the closing tag.
+                    // This ensures we retain the closing tag and don't break the html.
+                    let (preview, _) =
+                        preview.split_at(closing_tag_index + CLOSING_PARAGRAPH_TAG.len() + 1);
+                    if preview.chars().count() > MAX_PREVIEW_CHARS && preview.contains('\n') {
+                        match preview.find('\n') {
+                            Some(newline_index) => preview.split_at(newline_index).0.to_string(),
+                            None => unreachable!("Previous logic prevents this panic"),
+                        }
+                    } else {
+                        preview.to_string()
+                    }
+                }
+                None => unreachable!("Previous logic prevents this panic"),
+            }
```

Comment:
Rather than using a `None` branch with unreachable, this is what `unwrap` or `expect` are good for, e.g.

```rust
            let closing_tag_index = preview.find(CLOSING_PARAGRAPH_TAG).expect("closing tag out of range");
```